torch/distributed/_shard/sharding_spec/chunk_sharding_spec_ops/softmax.py

```diff
@@ -14,14 +14,14 @@ def sharded_softmax(types, args=(), kwargs=None):
     pg = input._process_group
     dim = kwargs['dim']
     sharding_dim = input.sharding_spec().dim
-    ndims = len(input.size())
+    ndims = input.dim()
     if dim == sharding_dim or dim + ndims == sharding_dim or sharding_dim + ndims == dim:
         exp = torch.exp(input.local_tensor())
         exp_sum = exp.sum(dim=dim).unsqueeze(dim=dim)
         exp_sum = torch.distributed.nn.functional.all_reduce(exp_sum, group=pg)
         smax = torch.div(exp, exp_sum)
     else:
-        smax = torch.nn.functional.softmax(input.local_tensor())
+        smax = torch.nn.functional.softmax(input.local_tensor(), dim=dim)
```

Comment:
Thanks for fixing this! Can we extend `test_softmax.py` to ensure there was a unit test covering this?

Reply:
Sure, maybe I can move this into `test_softmax.py`?

Reply:
Looks like `test_softmax.py` already contains this but it somehow use the default dim as the sharding dim so we didn't find out the corner case.